### DISCUSSION TITLE
home-manager.users.<user>.programs.bemenu does not exist

### DISCUSSION THREAD

**Discussion by moleculezz** [nix-community/stylix#287] [Q&A] [answered]
Hi,

I'm trying to get stylix working. But I get the following error when trying to build.

```
error: The option `home-manager.users.gd.programs.bemenu' does not exist. Definition values:
       - In `/nix/store/w48ijyc7k5g1h9yyz0x8ar655avpb8wp-modules/bemenu/hm.nix':
           {
             _type = "if";
             condition = true;
             content = {
               settings = {
           ...
```

I'm running Hyprland, but with rofi, not bemenu. And this error only pops up when I add stylix to my flake,

``` 
inputs....
stylix = {
      url = "github:danth/stylix";
      inputs.nixpkgs.follows = "nixpkgs";
      inputs.home-manager.follows = "home-manager";
    };

outputs...

stylix.nixosModules.stylix
stylix.image = ./wallpaper.jpg;
```

I don't understand why this is happening.
Am I doing something wrong here?

**Comment by danth** [comment]
Try updating Home Manager. You might be using an old version where the bemenu options don't exist, so when Stylix goes to check whether you have bemenu installed to apply the theme, we get this error.

**Reply by moleculezz** [reply]
I'm on 23.11 branch and I regularly do `nix flake update`
```
nixpkgs.url = "github:nixos/nixpkgs/nixos-23.11";
home-manager.url = "github:nix-community/home-manager/release-23.11";
home-manager.inputs.nixpkgs.follows = "nixpkgs";
```
So, I should switch to unstable?

**Reply by danth** [reply, marked as answer]
Either switching nixpkgs and Home Manager to unstable, or switching Stylix to `release-23.11` should fix it.
